### PR TITLE
Integrate ground data model objects into client

### PR DESCRIPTION
**Pull request by seanlobo**
`client.py` now returns data model objects instead of the json response from the server when applicable. The `get_*_history` and `get_*_latest_versions` methods are exceptions, as it those return lists/dicts of data instead of objects. Also cleaned up a bit of the style, removed some typos, changed method names from camel case to underscores and modified some default dictionary/lists to be None and initialized in the method.

~This pr doesn't include any tests yet, I can make a separate one when those are finished or add them here later.~ Tests have been added

### CODE REVIEW

File: python/ground/client.py [hunk 1]
```diff
@@ -55,54 +58,67 @@ def _getRichVersionJson(self, reference, reference_parameters, tags, structure_v
...
+                            to_node_version_start_id,
+                            from_node_version_end_id=-1,
+                            to_node_version_end_id=-1,
+                            reference=None,
+                            reference_parameters=None,
+                            tags=None,
+                            structure_version_id=-1,
+                            parent_ids=None):
+
+        if reference_parameters is None:
```
Comment (vsreekanti) [inline python/ground/client.py]: Why is this being done here? This should be done in `_get_rich_version_json`.

File: python/ground/client.py [hunk 2]
```diff
@@ -114,193 +130,284 @@ def createEdgeVersion(self,
...
-    def createGraphVersion(self,
-            graph_id,
-            edge_version_ids,
-            reference=None,
-            reference_parameters={},
-            tags={},
-            structure_version_id=-1,
-            parent_ids=[]):
+    def create_graph(self, source_key, name, tags=None):
+        if tags is None:
```
Comment (vsreekanti) [inline python/ground/client.py]: Same here.

Comment (vsreekanti) [reply python/ground/client.py]: And it seems with all of the `create` methods.

File: python/ground/common/model/core/edge.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+from ..version.item import Item
+
+
+class Edge(Item):
```
Comment (vsreekanti) [inline python/ground/common/model/core/edge.py]: There should also be a constructor that takes in the actual fields. The client methods should be able to take in these objects as arguments to the `create` methods. e.g., `create_edge(edge)`.

File: python/ground/common/model/core/structure.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+from ..version.item import Item
```
Comment (vsreekanti) [inline python/ground/common/model/core/structure.py]: Is there no better way to do this? Okay if not, but seems clunky.

**Review by vsreekanti — CHANGES_REQUESTED** [review]
Mostly looks good! A few minor requests / changes and a couple questions. Will merge once those are addressed.

File: python/ground/client.py (hunk 1 above)
Comment (seanlobo) [reply python/ground/client.py]: Looks like in `_get_rich_version_json` the values are only added if they evaluate to `True` anyway, so we can just remove all of these as `None` and empty collections all evaluate to `False`

File: python/ground/common/model/core/edge.py (hunk 3 above)
Comment (seanlobo) [reply python/ground/common/model/core/edge.py]: Line 4 isn't defining a constructor, it's defining the `Edge` class as extending the `Item` class. On line 13 there's a classmethod `from_edge` which creates a new edge from a previous edge object (usage would be `new_edge = Edge.from_edge(id, old_edge)`. Should that be changed to use `old_edge`'s id (so usage could be `Edge.from_edge(old_edge)`, and they would both share the same id)?

File: python/ground/common/model/core/structure.py (hunk 4 above)
Comment (seanlobo) [reply python/ground/common/model/core/structure.py]: Updated these imports to be absolute

File: python/ground/client.py (hunk 2 above)
Comment (seanlobo) [reply python/ground/client.py]: Same as above

**Review by vsreekanti — APPROVED** [review]

**Pull request merged**